### PR TITLE
Fix building 18.01.1

### PR DESCRIPTION
**Pull request by brobeson**
- Since January 2018, Eigen moved from Bitbucket to Gitlab. This pull request fixes the download URL and MD5 for Eigen.
- Geotranz and the geotiff data moved, too.
- Bzip2 also moved.
- I added a Dockerfile and devcontainer files for building externpro in a development container.

@swflb, we need this prior to getting a devcontainer to work for ovsrpro.

### CODE REVIEW

File: .devcontainer/Dockerfile [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+
+USER 0
+RUN set -eux; \
+  yum --assumeyes --quiet update; \
+  yum install --assumeyes \
+    epel-release \
+    # This repository gives us more modern versions of some software, such as Git.
+    https://packages.endpoint.com/rhel/7/os/x86_64/endpoint-repo-1.7-1.x86_64.rpm; \
+  yum install --assumeyes \
+    cmake3 \
```
Comment (swflb) [inline .devcontainer/Dockerfile:15]: which version of cmake3 does this install?

File: .devcontainer/Dockerfile [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+    git \
```
Comment (swflb) [inline .devcontainer/Dockerfile:16]: in the past this has installed git 1.8. terrible.

Comment (swflb) [reply .devcontainer/Dockerfile:16]: consider installing `rh-git218`

**Review by swflb — APPROVED** [review]

File: .devcontainer/Dockerfile [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+    gtk2-devel \
+    make \
+    mesa-libGL-devel \
+    mesa-libGLU-devel \
+    ncurses-devel \
+    ninja-build \
+    wget; \
```
Comment (swflb) [inline .devcontainer/Dockerfile:23]: I am surprised this list is so short

File: .devcontainer/Dockerfile (hunk 1 above)
Comment (brobeson) [reply .devcontainer/Dockerfile:15]: 3.17.3

File: .devcontainer/Dockerfile [hunk 4]
```diff
@@ -0,0 +1,41 @@
+FROM centos/devtoolset-6-toolchain-centos7:latest
```
Comment (bnhhughes) [inline .devcontainer/Dockerfile:1]: devtoolset-6 is deprecated now as well - I don't think it's even in the repos anymore.  Are you able to move to a newer one or is that not feasible?

**Review by bnhhughes — APPROVED** [review]

File: .devcontainer/Dockerfile (hunk 4 above)
Comment (swflb) [reply .devcontainer/Dockerfile:1]: @bnhhughes we want to change as little as possible for now. we will make some updates and then migrate to a newer gcc.

File: .devcontainer/Dockerfile (hunk 2 above)
Comment (brobeson) [reply .devcontainer/Dockerfile:16]: See the comment on line 12. This installs git 2.24.1. I can check for a more official RH package, but my container can't access yum repositories at the moment. I can double check later.

File: .devcontainer/Dockerfile (hunk 4 above)
Comment (brobeson) [reply .devcontainer/Dockerfile:1]: @bnhhughes Do you know when they plan to pull the plug entirely on devtoolset-6?

File: .devcontainer/Dockerfile (hunk 2 above)
Comment (brobeson) [reply .devcontainer/Dockerfile:16]: I can't get Git to be available if I install rh-git218 packages. I'd prefer to stick with epel if we could, but whichever rh-git218 package I try, the git command is not available.

File: .devcontainer/Dockerfile (hunk 4 above)
Comment (bnhhughes) [reply .devcontainer/Dockerfile:1]: I think they already have - we were using it in a chroot for a while and last time we went to build one from scratch we couldn't even pull down devtoolset-6 packages.  If you go to the Software Collections page for devtoolset-6 it points you to the repo, but the repo starts with devtoolset-7 (http://mirror.centos.org/centos/7/sclo/x86_64/rh/Packages/d/).  We've moved to devtoolset-9 which gives you GCC 9.3.1

File: .devcontainer/Dockerfile [hunk 5]
```diff
@@ -20,6 +20,7 @@ RUN set -eux; \
     mesa-libGLU-devel \
     ncurses-devel \
     ninja-build \
+    python-devel \
```
Comment (swflb) [inline .devcontainer/Dockerfile:23]: I approve but why is this necessary?

**Review by swflb — COMMENTED** [review]
#Approved

File: .devcontainer/Dockerfile (hunk 5 above)
Comment (brobeson) [reply .devcontainer/Dockerfile:23]: Its in the commit message. Boost Python fails to build without it. I thought this was in my pull request but realized later that I overlooked committing and pushing.
